src/v/cluster/partition.cc

```diff
@@ -57,7 +57,8 @@ partition::partition(
         }
 
         bool has_rm_stm = !_raft->log_config().is_compacted()
-                          && (_is_tx_enabled || _is_idempotence_enabled);
+                          && (_is_tx_enabled || _is_idempotence_enabled)
+                          && model::controller_ntp != _raft->ntp();
```

Comment:
what other internal topics do you have in mind, consumer groups?

Reply:
see model/namespace.h. consumer groups is one. @BenPope what about schema registry? I feel like there is another, but maybe not.

Reply:
> @BenPope what about schema registry?

The `_schemas` topic is just a normal topic in the `kafka` namespace.

Reply:
I've checked and we used to create rm_stm for the consumer group topics, fixed it. Since schema registry works via Kafka API so we don't need to exclude it.